### PR TITLE
TSPD exports support

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: packages/spec-lib/package.json [hunk 1]
```diff
@@ -17,7 +17,10 @@
   "main": "dist/index.js",
   "tspMain": "lib/lib.tsp",
   "exports": {
-    ".": "./dist/src/index.js"
+    ".": {
```
Comment (chrisradek) [inline packages/spec-lib/package.json:20]: Surprised this didn't have a `types` field before. Maybe needs one for `./dist/src/index.d.ts`?

Actually looks like the `main` export also should be `./dist/src/index.js` (I guess node.js prefers `exports` over `main`)

File: packages/tspd/src/gen-extern-signatures/gen-extern-signatures.ts [hunk 2]
```diff
@@ -2,32 +2,74 @@ import {
...
+}
 export async function generateExternSignatures(
   host: CompilerHost,
   libraryPath: string,
 ): Promise<readonly Diagnostic[]> {
   const diagnostics = createDiagnosticCollector();
   const pkgJson = await readPackageJson(host, libraryPath);
-  if (!pkgJson.tspMain) {
-    throw new Error("Must have a tspMain with decorator declaration.");
+  if (!pkgJson.exports) {
```
Comment (chrisradek) [inline packages/tspd/src/gen-extern-signatures/gen-extern-signatures.ts:31]: Do we want to drop support for just using `tspMain`? I'm fine with that given this is only used internally but probably need to double check that our scaffolding is updated to use `exports`.

Alternatively - check for `tspMain` and set it as the `"."` key in the `exportsMap`?

Comment (timotheeguerin) [reply packages/tspd/src/gen-extern-signatures/gen-extern-signatures.ts:31]: yeah wanted to drop it as I think we should recommend people use `exports` asnd having tspd do that would be good

File: packages/spec-lib/package.json (hunk 1 above)
Comment (timotheeguerin) [reply packages/spec-lib/package.json:20]: typescript will infer the typez auto from import or default

Comment (timotheeguerin) [reply packages/spec-lib/package.json:20]: but yeah made an error copying here

**Review by chrisradek — APPROVED** [review]
Imagine there will be some work in typespec-azure once this is merged, but looks good!

**Pull request merged**